modules/dialog/dlg_timer.c

```diff
@@ -840,7 +840,17 @@ void dlg_options_routine(unsigned int ticks , void * attr)
 		shm_free(it);
 		it = curr;
 
-		init_dlg_term_reason(dlg,"Ping Timeout",sizeof("Ping Timeout")-1);
+		if (dlg->legs[DLG_CALLER_LEG].reply_received == DLG_PING_FAIL)
+			init_dlg_term_reason(dlg,"Downstream Ping Timeout",sizeof("Downstream Ping Timeout")-1);
```

Comment:
@lemenkov , in the dialog module, the convention is the call flows from caller to callee, so caller -> callee is a downstream direction and callee -> caller is upstream direction ; sew https://opensips.org/html/docs/modules/3.2.x/dialog.html#pv_DLG_dir
So if the pinging failed on the caller side, it should be `Upstream Ping Timeout`, so the other way around :)
Or, if you want to keep it simple and 100% clear, better use `Caller Ping Timeout` .